src/Microsoft.OData.Client/ALinq/ExpressionWriter.cs

```diff
@@ -413,7 +415,7 @@ internal override Expression VisitMethodCall(MethodCallExpression m)
 
                     if (m.Method.Name != "GetValue" && m.Method.Name != "GetValueAsync")
                     {
-                        if (this.parent != null)
+                        if (m.Object != null && (this.InSubScope || this.parent?.NodeType == ExpressionType.Call))

```

Comment:
Could you explain this change?

Reply:
This part actually relates to the fix in https://github.com/OData/odata.net/pull/2572 was not quite right. You can have a parent that is not representing something in front of the slash. The correct detection is 
`if (m.Object != null && (this.InSubScope || this.parent?.NodeType == ExpressionType.Call))`
As this is what is being checked before writing $it in VisitInputReferenceExpression.
I have modified the test around this to reflect this scenario.